tests/utils/streams/test_kafka.py

```diff
@@ -88,3 +89,44 @@ def test_consumer(topic: str) -> None:
 
     with pytest.raises(RuntimeError):
         consumer.close()
+
+
+def test_commit_log_consumer(topic: str) -> None:
+    # XXX: This would be better as an integration test (or at least a test
+    # against an abstract Producer interface) instead of against a test against
+    # a mock.
+    commit_log_producer = FakeConfluentKafkaProducer()
+
+    consumer = KafkaConsumerWithCommitLog(
+        {
+            **configuration,
+            "auto.offset.reset": "earliest",
+            "enable.auto.commit": "false",
+            "enable.auto.offset.store": "true",
+            "enable.partition.eof": "true",
+            "group.id": "test",
+            "session.timeout.ms": 10000,
+            "default.topic.config": {  # XXX: Need to upgrade confluent-kafka
+                "auto.offset.reset": "earliest",
+            }
+        },
+        commit_log_producer,
+        'commit-log',
+    )
```

Comment:
This test highlights another design drawback of the subclass approach — it forces this test to be an integration test. An alternative approach here would be to implement the `KafkaConsumerWithCommitLog` as a wrapper around a `Consumer[TopicPartition, int, bytes]`, which would allow for us to avoid interacting with the actual Kafka broker here. I think that is alright, since the `KafkaConsumer` is already integration tested in the test above.

The issue with this is that the consumer group ID would have to be duplicated, since it would be part of the `KafkaConsumer` configuration (not available via the `Consumer` interface) as well as passed as a separate constructor argument here.